### DISCUSSION TITLE
Helm Chart for Kubernetes Deployments

### DISCUSSION THREAD

**Discussion by jslay88** [teslamate-org/teslamate#1932] [Ideas]
I have a Helm chart that I put together that deploys TeslaMate to Kubernetes clusters. However, I have slapped this together for my implementation, and probably isn't quite ready for "production". I am willing to share what I have, but I fear I do not have the time for a while, to properly finish building out this chart to the standards that one would deem acceptable for release. 

Thoughts? Ideas?

**Comment by fawaf** [comment]
definite +1. similar to https://github.com/teslamate-org/teslamate/issues/313. that repo is archived now since it moved to k8s-at-home, but k8s-at-home also is discontinued, so there is no "real" helm chart anymore. would be great to have an officially supported helm chart from teslamate.

**Reply by fawaf** [reply]
happy to help maintain it fwiw. :)

**Comment by brianmay** [comment]
I while ago I created helm chart:

* https://github.com/brianmay/charts/tree/main/charts/teslamate
* https://github.com/brianmay/charts/tree/main/charts/teslamate-grafana

I haven't used or updated in a while though, I found NixOS better met my needs.

**Reply by fawaf** [reply]
> I found NixOS better met my needs.

you mean running nixos without running k8s? or you run k8s using nix? (with teslamate)

**Reply by brianmay** [reply]
I am running nixos without k8s here.

**Comment by jelmervdm** [comment]
I don't know anything about creating helm charts (although I use them), but I have a couple yaml files I used for my deployment on k8s. I have a postgres database in CloudNative PG, use HA for mqtt and have a separate grafana instance to which I added the teslamate CNPG database to as a datasource.

Happy to sanitize them and share those. I generally prefer straight yaml files from the maintainer I can copy / paste / edit vs a third party helm chart from someone I don't know :).

**Reply by brianmay** [reply]
I myself am gradually falling into the "writing Kubernetes yaml files with go templates is a stupid idea". Unfortunately, helm still is the one standard for sharing Kubernetes config and avoiding reinventing the wheel. But it suffers serious limitations if doing anything that is vaguely complicated.

If I was creating a new system from scratch, would seriously look at at alternative. e.g.
* https://kubenix.org/
* https://www.pulumi.com/kubernetes/ (note example just uses a helm chart - suspect you can create all resources using pulumi without helm).

**Reply by fawaf** [reply]
> writing Kubernetes yaml files with go templates is a stupid idea

curious if you mean specifically go templates that suck or templating in general? have you heard of/used kluctl before (https://kluctl.io/)? it doesn't use go templating for a good reason and does seem better than helm imo.

**Reply by brianmay** [reply]
Haven't heard of kluctl. Thanks for the reference. Will investigate.
